lib/workers/repository/init/inherited.ts

```diff
@@ -99,5 +101,42 @@ export async function mergeInheritedConfig(
       'Removed global config from inherited config.',
     );
   }
-  return mergeChildConfig(config, filteredConfig);
+
+  if (is.nullOrUndefined(filteredConfig.extends)) {
+    return mergeChildConfig(config, filteredConfig);
+  }
+
+  let returnConfig = filteredConfig;
+
+  // Decrypt after resolving, in case the preset contains npm authentication
+  logger.debug('Resolving presets found in inherited config');
+  const resolvedConfig = await decryptConfig(
+    await resolveConfigPresets(returnConfig, config, config.ignorePresets),
+    config.repository,
+  );
+  logger.trace({ config: resolvedConfig }, 'Resolved inherited config');
+  const validationRes = await validateConfig('inherit', resolvedConfig);
+  if (validationRes.errors.length) {
+    logger.warn(
+      { errors: validationRes.errors },
+      'Found errors in presets inside the inherited configuration.',
+    );
+    throw new Error(CONFIG_VALIDATION);
+  }
+  if (validationRes.warnings.length) {
+    logger.warn(
+      { warnings: validationRes.warnings },
+      'Found warnings in presets inside the inherited configuration.',
+    );
+  }
+
+  returnConfig = removeGlobalConfig(resolvedConfig, true);
+  if (!dequal(resolvedConfig, returnConfig)) {
+    logger.debug(
+      { inheritedConfig: resolvedConfig, filteredConfig: returnConfig },
+      'Removed global config from inherited config presets.',
+    );
```

Comment:
Inherited config does allow some fields which were previously considered "global", such as `onboarding`. Please check that we don't log in this case

Reply:
Even today `onboarding` is not removed as it has `inheritConfigSupport=true`.